atc/exec/build/repository_test.go

```diff
@@ -0,0 +1,77 @@
+package build_test
+
+import (
+	. "github.com/concourse/concourse/atc/exec/build"
+	"github.com/concourse/concourse/atc/runtime/runtimefakes"
+	
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ArtifactRepository", func() {
+	var (
+		repo *Repository
+	)
+
+	BeforeEach(func() {
+		repo = NewRepository()
+	})
+
+	It("initially does not contain any artifacts", func() {
+		artifact, found := repo.ArtifactFor("first-artifact")
+		Expect(artifact).To(BeNil())
+		Expect(found).To(BeFalse())
+	})
+
+	Context("when a artifact is registered", func() {
+		var firstArtifact *runtimefakes.FakeArtifact
+
+		BeforeEach(func() {
+			firstArtifact = new(runtimefakes.FakeArtifact)
+			repo.RegisterArtifact("first-artifact", firstArtifact)
+		})
+
+		Describe("ArtifactFor", func() {
```

Comment:
This test file probably be shortened by just testing the "multiple artifacts" route, since having one artifact is implicitly checked by having two.